credentialdigger/models/snippet_model/snippet_model.py

```diff
@@ -76,17 +77,24 @@ def analyze(self, discovery):
             # since either the snippet is empty or there is just one word
             return True
 
-        if len(data) == 2:
-            # No need to pre-process data since we have only two words
-            # Predict if the string `word1 + ' ' + word2` is a false positive
-            label = self.model.predict(
-                data[0] + ' ' + data[1])[0]  # 0=label, 1=probability
+        # Extract the committed secret
+        index_of_value = self._label_preprocess(data)
+        
+        # Classify as a 'Leak' if this is a private key.
+        if self._check_private_key(data):
+            return False
+
+        # We ignore snippets that look like regular phrases with
+        # no assignment
+        if not any(not c.isalnum() and c not in ' _.,?!/' for c in raw_data):
+            return True
+
+        input_text = data[index_of_value[1]]
```

Comment:
place a comment here explaining why you do this apparently weird transformation: why position 1 of `index_of_value`? Why is data needed?

Reply:
`input_text` is a misleading name: this is not an input, this is "a bet", i.e., the value that the extractor identify as the secret/token/password value

Reply:
> place a comment here explaining why you do this apparently weird transformation: why position 1 of index_of_value? Why is data needed?

- data is what we are left with after pre-processing the snippet
   - if we have `raw_data='pass_word="123"`, then data will be `[PassWord, 123]`
- we use only position 1 because the new model (and then old one too, that's been trained using the new datasets (to be eventually released)) only takes 1 value as input, contrary to the old strategy of taking a couple (key,value). So we only retrieve the value (index 1) and ignore the variable name (index 0).